rotkehlchen/inquirer.py

```diff
@@ -1157,13 +1162,14 @@ def find_yearn_price(
         # Get the price per share from the yearn contract
         contract = EvmContract(
             address=token.evm_address,
-            abi=ethereum.node_inquirer.contracts.abi('YEARN_VAULT_V2'),
+            abi=ethereum.node_inquirer.contracts.abi(vault_abi),
             deployed_block=0,
         )
         try:
             price_per_share = contract.call(ethereum.node_inquirer, 'pricePerShare')
         except (RemoteError, BlockchainQueryError) as e:
-            log.error(f'Failed to query pricePerShare method in Yearn v2 Vault. {e!s}')
+            version = 'v2' if vault_abi == 'YEARN_VAULT_V2' else 'v3'
+            log.error(f'Failed to query pricePerShare method in Yearn {version} Vault. {e!s}')
```

Comment:
I believe is better to state the token here rather than the version. You also avoid that ugly if check. you can write

```suggestion
            log.error(f'Failed to query pricePerShare method for Yearn vault {token}. {e!s}')
```